src/com/company/Main.java

```diff
@@ -8,82 +8,60 @@
 
 
 public class Main {
-    public static void main(String[] args){
-
-        String txtfile = args[0];
-        File file = new File(txtfile);
-        checkforsupport(file);
-        checkforexistance(file);
+    final static String FILE_TYPE=".txt";
+    final static String SPECIAL_CHAR_REMOVAL_REGEX="[^a-zA-Z0-9@]";
+    final static String SINGLE_SPACE=" ";
+    public static void main(String[] args) {
+
+        if (args.length != 2) {
+            System.out.println("Exiting, since no file path and word to search for are provided!");
+            return;
+        }
+        String txtFilePath = args[0];
         String searchword = args[1];
-        System.out.println("processing................");
-        String data = null;
-        data = readFileAsString(txtfile);
-        avoidspecialchar(data);
-        searchtheword(data,searchword);
-
+        File file = new File(txtFilePath);
 
+        if (!file.exists() || !isFileFormatSupported(file)) {
+            System.out.println("File is not valid");
+            return;
+        }
+        if (file.length() == 0) {
+            System.out.println("The file does not contains any data ");
+            return;
+        }
+        System.out.println("Processing................");
+        String data = readFileAsString(txtFilePath);
+        searchTheWord(data, searchword);
     }
 
-
-        public static String readFileAsString(String fileName)
-             {
-                String data = "";
-              try {
-                   data = new String(Files.readAllBytes(Paths.get(fileName)));
-              } catch (IOException e) {
-                  e.printStackTrace();
-              }
-              return data;
+    public static String readFileAsString(String fileName) {
+        String data =null;
+        try {
+            data = new String(Files.readAllBytes(Paths.get(fileName)));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return data.replaceAll(SPECIAL_CHAR_REMOVAL_REGEX, SINGLE_SPACE);
```

Comment:
There is a possibility of null pointer exception here. What happens if there is no data in file and program execution continues?

Reply:
resolved with writing replaceall method within try block